R/dataframe.R

```diff
@@ -182,24 +182,53 @@ as_data_frame.matrix <- function(x, ...) {
   as_data_frame(as.data.frame(x))
 }
 
-#' Convert row names to an explicit variable.
+#' Converting between rownames and a column
+#'
+#' add_rownames: Convert row names to an explicit variable.
 #'
 #' @param df Input data frame with rownames.
 #' @param var Name of variable to use
 #' @export
+#' @rdname add_rownames
 #' @examples
 #' tbl_df(mtcars)
 #'
-#' add_rownames(mtcars)
+#' mtcars_tbl <- add_rownames(mtcars)
 add_rownames <- function(df, var = "rowname") {
   stopifnot(is.data.frame(df))
 
+  if (var %in% colnames(df) ) {
+    stop(paste("There is a column named", var, "already!"))
+  }
+
   rn <- as_data_frame(setNames(list(rownames(df)), var))
-  rownames(df) <- NULL
+  rownames(rn) <-rn[[var]]
```

Comment:
It is `row.names = c(NA, 32L)` outputted by dput(add_rownames(...)). But `rownames( add_rownames(...) )` will return `as.character(1:32)`. I'm not sure what it be affected. 

So far. `tbl_df` keeps the row names, while `add_rownames` update it as a character number vector, `as.character(seq_len(NROW(x))`. Kind of inconsistent. So what's your suggestion?